tubemq-client/src/main/java/org/apache/tubemq/client/producer/SimpleMessageProducer.java

```diff
@@ -335,11 +336,13 @@ private void checkMessageAndStatus(final Message message) throws TubeClientExcep
     }
 
     private byte[] encodePayload(final Message message) {
-        final byte[] payload = message.getData();
-        final String attribute = message.getAttribute();
+        byte[] payload = message.getData();
+        String attribute = message.getAttribute();
         if (TStringUtils.isBlank(attribute)) {
-            return payload;
+            attribute = "";
         }
+        attribute = attribute + "," + TokenConstants.TOKEN_COMPRESS_TYPE + producerConfig.getCompressionType().name();
```

Comment:
The attribute item in the attribute needs to be added in a=b, c=d mode and has a fixed format. It is best to call the method of Message to add the attribute item in this place, instead of seeming to manually stitch together to avoid incompatibility after adjusting the encoding